src/families/tron/AccountBalanceSummaryFooter.js

```diff
@@ -1,118 +1,130 @@
+    [freeLimit, gainedLimit, gainedUsed, freeUsed],
   );
 
-  const formatedEnergy = energy;
+  const onCloseModal = useCallback(() => {
+    setInfoName(undefined);
+  }, []);
+
+  const onPressInfoCreator = useCallback(
+    (infoName: InfoName) => () => setInfoName(infoName),
+    [],
+  );
+
+  const isModalOpen = useMemo(() => !!infoName, [infoName]);
+
+  const data = useMemo(() => (infoName ? infoCandidates[infoName] : []), [
+    infoName,
+    infoCandidates,
+  ]);
 
-  const formatedBandwidth = freeLimit + gainedLimit - gainedUsed - freeUsed;
+  if (!account.tronResources) return null;
 
   return (
     <ScrollView
       horizontal
       showsHorizontalScrollIndicator={false}
       style={styles.root}
     >
-      <TouchableOpacity
-        onPress={() => {
-          /** @TODO redirect to info modal */
-        }}
-        style={styles.balanceContainer}
-      >
-        <View style={styles.balanceLabelContainer}>
-          <LText style={styles.balanceLabel}>
-            <Trans i18nKey="account.availableBalance" />
-          </LText>
-          <Info size={12} color={colors.grey} />
-        </View>
-        <LText semiBold style={styles.balance}>
-          {spendableBalance}
-        </LText>
-      </TouchableOpacity>
-      <TouchableOpacity
-        onPress={() => {
-          /** @TODO redirect to info modal */
-        }}
-        style={styles.balanceContainer}
-      >
-        <View style={styles.balanceLabelContainer}>
-          <LText style={styles.balanceLabel}>
-            <Trans i18nKey="account.tronPower" />
-          </LText>
-          <Info size={12} color={colors.grey} />
-        </View>
-
-        <LText semiBold style={styles.balance}>
-          {tronPower}
-        </LText>
-      </TouchableOpacity>
-      <TouchableOpacity
-        onPress={() => {
-          /** @TODO redirect to info modal */
-        }}
-        style={styles.balanceContainer}
-      >
-        <View style={styles.balanceLabelContainer}>
-          <LText style={styles.balanceLabel}>
-            <Trans i18nKey="account.bandwidth" />
-          </LText>
-          <Info size={12} color={colors.grey} />
-        </View>
-
-        <LText semiBold style={styles.balance}>
-          {formatedBandwidth || "–"}
-        </LText>
-      </TouchableOpacity>
-      <TouchableOpacity
-        onPress={() => {
-          /** @TODO redirect to info modal */
-        }}
-        style={styles.balanceContainer}
-      >
-        <View style={styles.balanceLabelContainer}>
-          <LText style={styles.balanceLabel}>
-            <Trans i18nKey="account.energy" />
-          </LText>
-          <Info size={12} color={colors.grey} />
-        </View>
-        <LText semiBold style={styles.balance}>
-          {formatedEnergy || "–"}
-        </LText>
-      </TouchableOpacity>
+      <InfoModal isOpened={isModalOpen} onClose={onCloseModal} data={data} />
+
+      <InfoItem
+        title={t("account.availableBalance")}
```

Comment:
i don't think using and connecting the translation function here is necessary you could just use a Trans component thus limiting higher order functions usage.
like so `title={<Trans i18nKey="account.availableBalance" />}`
if theres an issue on typing instead of using string you can use React$Node for flow. 

Alternatively you can send the translation key instead of the translated string and use a Trans component inside InfoItem :man_shrugging:

Reply:
if you upgrade react-i18n, they have handy hook called "useTranslation". I tried to use that but apparently, it doesn't exist in this version so I went with HOC. It is just a temporary solution.

Reply:
`<Trans` should work regardless of react-i18n version and we generally prefer it

Reply:
I prefer `const {t} = useTranslation();` in general because it's simply returns string (also the type).

Reply:
https://react.i18next.com/latest/trans-component#important-note